shared/constants/types/flux.js

```diff
@@ -19,4 +19,4 @@ export type AsyncAction = (dispatch: Dispatch, getState: GetState) => ?Promise<*
 export type Dispatch = (action: AsyncAction | Action) => ?Promise<*>
 
 export type TypedAsyncAction<A> = (dispatch: TypedDispatch<A>, getState: GetState) => ?Promise<*>
-export type TypedDispatch<A> = (action: TypedAsyncAction<A> | A) => ?Promise<*>
+export type TypedDispatch<-A> = (action: TypedAsyncAction<A> | A) => ?Promise<*>
```

Comment:
Had a discussion with mgood, here's our argument for why this is contravariant:

Let’s say we have a type `T` that represents an enum of actions `A | B | C`
so `type T = A | B | C`

now we have type `T'`

`type T' = A | B | C | D`

is type `T` a subtype of `T'`? Yes. aka (`T <: T'`)

So is `TypedDistpatch<T'> <: TypedDispatch<T>` true? Yes.

therefore the types are contravariant